src/unix/linux/utils.rs

```diff
@@ -34,15 +35,12 @@ pub(crate) struct FileCounter(File);
 
 impl FileCounter {
     pub(crate) fn new(f: File) -> Option<Self> {
-        unsafe {
-            if let Ok(ref mut x) = REMAINING_FILES.lock() {
-                if **x > 0 {
-                    **x -= 1;
-                    return Some(Self(f));
-                }
-                // All file descriptors we were allowed are being used.
-            }
+        if REMAINING_FILES.load(Ordering::SeqCst) > 0 {
+            REMAINING_FILES.fetch_sub(1, Ordering::SeqCst);
```

Comment:
That's the same issue here. Between the `load` and the `fetch_sub`, an update could have happened.

Reply:
Whoops, fixed.